src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,159 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import createFocusTrap from 'focus-trap';
+import { st, classes } from './Modal.st.css';
+import { TPAComponentsConsumer } from '../TPAComponentsConfig';
+import { ModalProps, ModalDefaultProps, ModalState } from './types';
+import { CloseIcon } from './Close';
+
+export class Modal extends React.Component<ModalProps, ModalState> {
+  static displayName = 'Modal';
+  static defaultProps: ModalDefaultProps = {
+    isOpen: false,
+    closeOnClickOutside: true,
+    minHeight: '150px',
+    maxHeight: '620px',
+    minWidth: 'auto',
+    maxWidth: '580px',
```

Comment:
So I think these props shouldn't be here, but actually in the Dialog component, since the `Modal` component shouldn't really care about style (except maybe for the background overlay color, etc)
In addition, instead of using javascript for these properties, consumers of the component can use css/stylable for this, just by passing a className, and giving a min/max height/width in css